x-pack/plugins/enterprise_search/server/routes/enterprise_search/public_url.test.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { MockRouter } from '../__mocks__/router.mock';
+
+jest.mock('../../lib/enterprise_search_config_api', () => ({
+  callEnterpriseSearchConfigAPI: jest.fn(),
+}));
+import { callEnterpriseSearchConfigAPI } from '../../lib/enterprise_search_config_api';
+
+import { registerPublicUrlRoute } from './public_url';
+
+describe('Enterprise Search Public URL API', () => {
+  const mockRouter = new MockRouter({ method: 'get' });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockRouter.createRouter();
+
+    registerPublicUrlRoute({
+      router: mockRouter.router,
+      config: {},
+      log: {},
+    } as any);
+  });
```

Comment:
Optional: I see you are doing it in other route tests: you can use core logger mock here instead of an empty object. Slightly better / more resilient, even if this specific route does not call `log`

```ts
import { loggingSystemMock } from 'src/core/server/mocks';
const logger = loggingSystemMock.createLogger();
```

Doing that + creating a 'real' `ConfigType` object would also allow you to get rid of the `as any` cast, which is always preferable even in test suites.